### PR TITLE
Fix/816

### PR DESCRIPTION
**Pull request by dangreen**
https://github.com/canvg/canvg/issues/816

- [x] handle nullish bounding box
- [x] text trim fix
- [x] font to string fix

### CODE REVIEW

File: src/Document/FilterElement.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ export default class FilterElement extends Element {
 			document
 		} = this;
 		const boundingBox = element.getBoundingBox(ctx);
```
Comment (gabelerner) [inline src/Document/FilterElement.ts]: when would this ever be missing?
`PathElement.getBoundingBox` is https://github.com/canvg/canvg/blob/master/src/Document/PathElement.ts#L110
which calls `path()` which always returns https://github.com/canvg/canvg/blob/master/src/Document/PathElement.ts#L106

what am i missing here?

Comment (gabelerner) [reply src/Document/FilterElement.ts]: image

even typescript says it's always there

Comment (dangreen) [reply src/Document/FilterElement.ts]: @gabelerner 
1) `PathElement` is base class for some other elements, where `getBoundingBox` can be redefined
2) `getBoundingBox` can return `null`, not `undefined`.

Comment (gabelerner) [reply src/Document/FilterElement.ts]: ah ok, i wish we can enforce it w/ types but kind of hard w/ polymorphism ... we can try to do some generic classes at some point like have a `PathElementWithBoundingBox` and a `PathElementWithoutBoundingBox` where they extend `PathElement<T>` where `T` is either a `BoundingBox` or `null` and then others can extend from there. for now will review rest of PR after i push out 3.0

Comment (dangreen) [reply src/Document/FilterElement.ts]: @gabelerner This fix should be in v3, there is text rendering fixes.

Comment (gabelerner) [reply src/Document/FilterElement.ts]: ok i'll push out 3.0.1 w/ it no problem ... let me finish reviewing

**Pull request merged**